CMakeLists.txt

```diff
@@ -399,6 +399,11 @@ else()
 endif()
 include(ThirdpartyToolchain)
 
+if (THRIFT_VERSION VERSION_LESS "0.11.0")
```

Comment:
This might need to be a build option to toggle rather than using the version number. I think in Thrift 0.11.0 there might be a way to build with `boost:shared_ptr` support

Reply:
`FORCE_BOOST_SMART_PTR` is the option to enforce boost in Thrift 0.11.0. Can users pass that via existing `PARQUET_CXXFLAGS`? The check `if defined(PARQUET_THRIFT_USE_BOOST) || defined(FORCE_BOOST_SMART_PTR)` in `thrift.h` examines that macro.

Reply:
Ref: https://thrift.apache.org/lib/cpp